app/helpers/cases_helper.rb

```diff
@@ -116,6 +116,12 @@ def action_button_for(event)
               case_sar_extensions_path(@case),
               id: 'action--remove-extended-deadline-for-sar',
               class: 'button-secondary', method: :delete
+    when :record_data_request
+      btn_type = @case.current_state == 'data_to_be_requested' ? 'secondary' : 'tertiary'
+      link_to 'Record data request',
+              new_case_data_request_path(@case),
+              id: 'action--record-data-request',
+              class: "button-#{btn_type}", method: :get
```

Comment:
This still POSTs to the next page - really annoying! But not worth spending time on IMO

Reply:
I will check this before merging to master as I have it definitely being s GET to the next page!

Reply:
Can you check this again - I cannot reproduce:
<img width="1340" alt="Screen Shot 2019-08-30 at 10 23 38" src="https://user-images.githubusercontent.com/47113046/64009711-42ac1580-cb10-11e9-9649-9457e51730f8.png">

Reply:
![image](https://user-images.githubusercontent.com/1161161/64010604-037ec400-cb12-11e9-9fb3-7580052e9e2e.png)
It's consistent in Chrome and Safari - click the "Record data request" button and hit reload in the browser (without the web inspector open) and it asks you to resubmit the form - header shows POST not GET. 

But as I said, the code in the view looks fine to me and IMO it's not worth spending time on investigating further just now - just annoying. Presumably there is some UJS overriding the data-method attribute somewhere

Reply:
Fixed - it's because in the helper I copy pasted a previous link/button, and it has `method: :get` set. This causes some interesting behaviour!

```
    when :record_data_request
      btn_type = @case.current_state == 'data_to_be_requested' ? 'secondary' : 'tertiary'
      link_to 'Record data request',
              new_case_data_request_path(@case),
              id: 'action--record-data-request',
              class: "button-#{btn_type}", method: :get
```

Removing the `method: :get` solves the problem!!